### ISSUE TITLE
[TypeScript] Missing type generation for a union that only resolves __typename

### ISSUE THREAD

**Issue by csoLs** [dotansimha/graphql-code-generator#1120]
**Describe the bug**
If you only query a union for the `__typename`, codegen doesn't generate a type but still references it. See example:
<img width="1429" alt="screen shot 2019-01-04 at 15 57 06" src="https://user-images.githubusercontent.com/9143012/50694231-9de93f00-1039-11e9-9534-f7506b9f6639.png">

**To Reproduce**
1. Write a schema containing a union
2. Query for the union `__typename` only

**Expected behavior**
A type with only the `__typename` should be generated. I.e. in this case the generated file should also contain:
```
export type UserUnion = {
  __typename: "OneUnion" | "TwoUnion";
}
```

**Comment by xanf** [comment]
@csoLs Just my 2c (I'm looking to fix this bug) - it would be awesome if your copy-paste text content in report, not just screenshots. That will simplify life just a bit, no offense. 

Being able to reproduce bug quickly is invaluable :)

**Comment by csoLs** [comment]
Schema:
```
schema {
  query: Query
}
type Query {
  user(id: ID!): User
}
type User {
  id: ID!
  union: UserUnion
}
union UserUnion = OneUnion | TwoUnion
type OneUnion {
  id: ID!
  name: String!
}
type TwoUnion {
  id: ID!
  name: String!
}
```

Document:
```
query findUser($userId: ID!) {
  user(id: $userId) {
    id
    union {
      __typename    
    }
  }
}
```
Config:
```
generates:
  client-types.ts:
    - typescript-common
    - typescript-client
```

**Comment by dotansimha** [comment]
Thanks for reporting if @csoLs ! 
We are filtering `__typename` and ignoring it from the fields list. It requires a more complex solution.

At the moment we are looking for refactoring the typescript packages, and use the same approach as `flow` plugins (use `Pick` and build a single type only, instead of namespace with multiple interfaces), and we also implemented there a solution for `__typename` fields).

**Comment by piotr-m-jurek** [comment]
Hi, I think this my question regards something else, but:
shouldn't the `__typename` field be required? It is even mentioned in the query. Shouldn't `__typename` be non-nullable by default?

**Comment by dotansimha** [comment]
Fixed in the recent refactor :)

**Comment by dotansimha** [comment]
Fix available in 1.0.0 👍
